### PR TITLE
Add Xor layer

### PR DESCRIPTION
**Pull request by Vimalakirti**

### CODE REVIEW

File: src/layer/xor.rs [hunk 1]
```diff
@@ -0,0 +1,46 @@
+use crate::basic_block::*;
```
Comment (ddkang) [inline src/layer/xor.rs:1]: Does this require that the inputs are in {0, 1}? Is this implicitly implied by the inputs?

Comment (Vimalakirti) [reply src/layer/xor.rs:1]: Probably the easiest way is we use a basicblock to perform the boolean check during model compilation.

File: src/basic_block/bool_check.rs [hunk 2]
```diff
@@ -0,0 +1,20 @@
+use super::{BasicBlock, Data, DataEnc, PairingCheck, ProveVerifyCache, SRS};
+use crate::util;
+use ark_bn254::{Fr, G1Affine, G1Projective, G2Affine, G2Projective};
+use ndarray::{arr0, azip, ArrayD, IxDyn};
+use rand::rngs::StdRng;
+
+// BooleanCheckBasicBlock is a basic block that checks if all elements are 0 or 1
```
Comment (ddkang) [inline src/basic_block/bool_check.rs]: Other PR

Comment (Vimalakirti) [reply src/basic_block/bool_check.rs]: see the response in #73

**Comment by liliatangxy** [comment]
lgtm minus the blinding comment in #74

**Comment by Vimalakirti** [comment]
> lgtm minus the blinding comment in #74

See response in https://github.com/uiuc-kang-lab/zk-torch/pull/73

**Review by liliatangxy — DISMISSED** [review]
good once we use And's bool check BB

**Review by ddkang — APPROVED** [review]

**Pull request merged**